src/os_crypto/md5_sha1/md5_sha1_op.c

```diff
@@ -86,3 +91,99 @@ int OS_MD5_SHA1_File(const char *fname, const char *prefilter_cmd, os_md5 md5out
 
     return (0);
 }
+
+#ifdef LIBSODIUM_ENABLED
+int OS_Hash_File(const char *fname, const char *prefilter_cmd, struct hash_output *file_output, int mode)
```

Comment:
Following along with the logic from above, SHA1 is supposed to be checked, but this function deals with MD5 and SHA256 only far as i see.

Reply:
It's been a long day, so I might be misunderstanding you. Bear with me and be gentle.
OS_MD5_SHA1_File was the original function. To make things easier for myself, OS_Hash_File was a copy of that (in the same file to, again, make things easier). Rather than deal with the directory naming issues and whatnot, I just wanted to get the functionality down.

Reply:
Sorry for the confusion, it looks to me as though these changes would depend on both functions - since MD5 and SHA1 are always checked, which seems a bit redundant in the first place, but very much so if we run 3 (4 if MD5 happens once in each function being called) hash passes. If in fact the new function is being called instead, then it won't deal with the SHA1 piece, which still  needs to be verified.

Reply:
I thought I had made it so only OS_Hash_File() gets called if libsodium support is added. I'll have to double check.